custom_components/ha_washdata/manager.py

```
@@ -2216,26 +2523,56 @@ def _on_cycle_end(self, cycle_data: dict[str, Any]) -> None:
-                cycle_energy_wh = float(np.sum(avg_p * np.abs(dt_h)))
+                valid = [(float(p[0]), float(p[1])) for p in power_data if isinstance(p[0], (int, float)) and isinstance(p[1], (int, float))]
+                if len(valid) >= 2:
+                    valid.sort(key=lambda x: x[0])
+                    ts = np.array([v[0] for v in valid])
+                    ps = np.array([v[1] for v in valid])
+                    dt_h = np.diff(ts) / 3600.0
+                    _MAX_GAP_H = 1.0  # skip segments longer than 1 hour
+                    mask = (dt_h > 0) & (dt_h <= _MAX_GAP_H)
+                    avg_p = (ps[:-1] + ps[1:]) / 2
+                    cycle_energy_wh = float(np.sum(avg_p[mask] * dt_h[mask]))
             except Exception:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)